### PR TITLE
fix(yield-calculator) fix pool inconsistancies

### PR DESCRIPTION
**Pull request by chrismaree**
This PR fixes an issue in the yield calculator for renBTC. Before this PR, the balancer graphQL query expects the `tokensList` in the pools had the synth token USDC token address in an array. This works for `yUSD-SEP20` and `yUSD-OCT20`. However for `uUSDrBTC-OCT` this order has flipped around. To accommodate any direction, this PR modifies the `YIELD_TOKENS` object to store a string-array key-value object to store relevant balancer lookups with the correct token ordering.

### CODE REVIEW

File: containers/Balancer.ts [hunk 1]
```diff
@@ -33,10 +33,23 @@ interface PoolTokenQuery {
...
 
-const YIELD_TOKENS = [
-  "0x81ab848898b5ffD3354dbbEfb333D5D183eEDcB5", // Sep20
-  "0xB2FdD60AD80ca7bA89B9BAb3b5336c2601C020b4", // Oct20
-];
+interface YIELD_TOKEN {
+  [key: string]: string[];
+}
+const YIELD_TOKENS: YIELD_TOKEN = {
+  "0x81ab848898b5ffD3354dbbEfb333D5D183eEDcB5": [
```
Comment (mrice32) [inline containers/Balancer.ts:40]: nit: why to lower case instead of to checksum address?

File: containers/Balancer.ts [hunk 2]
```diff
@@ -33,10 +33,23 @@ interface PoolTokenQuery {
...
+    "0x81ab848898b5ffD3354dbbEfb333D5D183eEDcB5".toLowerCase(),
+    "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48".toLowerCase(),
+  ], // yUSDETH-SEP20
+  "0xB2FdD60AD80ca7bA89B9BAb3b5336c2601C020b4": [
+    "0xb2fdd60ad80ca7ba89b9bab3b5336c2601c020b4".toLowerCase(),
```
Comment (mrice32) [inline containers/Balancer.ts:45]: What are the two elements in these arrays? Would it be clearer to make them key value pairs?

File: containers/Balancer.ts (hunk 1 above)
Comment (chrismaree) [reply containers/Balancer.ts:40]: the input to the graphQL query only accepts lower case addresses

File: containers/Balancer.ts (hunk 2 above)
Comment (nicholaspai) [reply containers/Balancer.ts:45]: Are they [0] = synth, [1] = collateral?

Comment (chrismaree) [reply containers/Balancer.ts:45]: In this, the key is the synthetic token and the value is the associated balancer pool pair in the correct order for querying. It cant be a key-value pair as sometimes the ordering in the query within balancer swaps direction. For some of our tokens, this is `[syntheticAddress,USDCAddress]` buy for others, like ren this is `[USDCAddress,syntheticAddress]`.

does that make sense?

File: features/yield/BalancerData.tsx [hunk 3]
```diff
@@ -60,7 +62,7 @@ const BalancerData = () => {
     return (
       <span>
         <Typography variant="h5" style={{ marginBottom: "10px" }}>
-          yUSD Pool Metrics{" "}
+          {tokenSymbol} Pool Metrics{" "}
```
Comment (nicholaspai) [inline features/yield/BalancerData.tsx:65]: Does this render something strange if the selected token is a non yield token?

File: containers/Balancer.ts (hunk 2 above)
Comment (chrismaree) [reply containers/Balancer.ts:45]: > Are they [0] = synth, [1] = collateral?

not always; this is the reason I had to add this mapping object like this. For the ren synthetic a query that works looks like this:
```
{
    pools(where: {tokensList: ["0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48","0x208d174775dc39fe18b1b374972f77ddec6c0f73"]}) {
      swapFee
      liquidity
      totalSwapVolume
      tokens {
        address
        balance
      }
      joinsCount
      exitsCount
      swapsCount
      totalShares
      shares(first:1000, where: { balance_gt: 0}) {
        userAddress {
          id
        }
        balance
      }
    }
  }
```
This has `[USDCAddress,syntheticAddress]` in the `where` clause. For the other yield tokens this would be swapped.

File: features/yield/BalancerData.tsx (hunk 3 above)
Comment (chrismaree) [reply features/yield/BalancerData.tsx:65]: Balancer data should only show up for a yield token. This page wont load if the token is not a yield token.

File: containers/Balancer.ts (hunk 2 above)
Comment (nicholaspai) [reply containers/Balancer.ts:45]: This makes sense. Would it be accurate (and more readable) to make the keys `token0` and `token1`?

File: features/yield/YieldCalculator.tsx [hunk 4]
```diff
@@ -58,8 +60,7 @@ const YieldCalculator = () => {
 
     // `yieldPerUnit` = (FACE/yUSD_PX)^(1/(365/DAYS_TO_EXP)) - 1,
     // where FACE = $1. More details: https://www.bankrate.com/glossary/a/apy-annual-percentage-yield/
-    const yieldPerUnit =
-      Math.pow(1 / _tokenPrice, 1 / (_daysToExpiry / DAYS_PER_YEAR)) - 1;
+    const yieldPerUnit = (1 - _tokenPrice) * (DAYS_PER_YEAR / _daysToExpiry);
```
Comment (nicholaspai) [inline features/yield/YieldCalculator.tsx:63]: @chrismaree @hlambur can you take a quick look at this? Chris thinks that we are calculating APY incorrectly

File: containers/Balancer.ts (hunk 2 above)
Comment (chrismaree) [reply containers/Balancer.ts:45]: Ye, sure. done. PTAL

File: containers/Balancer.ts (hunk 1 above)
Comment (chrismaree) [reply containers/Balancer.ts]: I cleaned this up a bit by pulling toLowerCase to the set function.

File: containers/Balancer.ts [hunk 5]
```diff
@@ -53,7 +73,7 @@ const useBalancer = () => {
   const [selectedSwapTokenAddress, setSelectedSwapTokenAddress] = useState<
     string | null
   >(null);
-  const [poolTokenList, setPoolTokenList] = useState<string[] | null>(null);
+  const [poolTokenList, setPoolTokenList] = useState<string[] | null>();
```
Comment (nicholaspai) [inline containers/Balancer.ts:76]: why did you change this?

**Review by nicholaspai — APPROVED** [review]
LGTM, I only had 1 question which might be really obvious but I don't udnerstand why you made!

File: containers/Balancer.ts (hunk 5 above)
Comment (chrismaree) [reply containers/Balancer.ts]: have reverted this
